docs/diagrams/kubernetes_workflow.md

```diff
@@ -0,0 +1,1155 @@
+        Watch->>KubeUnit: Event received
+        
+        KubeUnit->>KubeUnit: podRunningAndReady() check
+        
+        alt Pod Phase: Running or Pending
+            KubeUnit->>KubeUnit: Check PodReady condition
+            alt PodReady == True
+                Watch-->>KubeUnit: Pod ready event
+            else ContainersReady == False
+                KubeUnit->>KubeUnit: Check container status
+                alt ImagePullBackOff
+                    KubeUnit->>KubeUnit: Retry check (max 3 times)
+                    alt Retries exhausted
+                        Watch-->>KubeUnit: Error: ErrImagePullBackOff
+                    end
+                end
+            end
+        else Pod Phase: Failed
+            Watch-->>KubeUnit: Error: ErrPodFailed
+        else Pod Phase: Succeeded
+            Watch-->>KubeUnit: Error: ErrPodCompleted
+        else Pod Deleted
+            Watch-->>KubeUnit: Error: NotFound
+        end
+    end
+
+    alt Pod Ready
+        KubeUnit->>KubeUnit: Store pod object
+        KubeUnit-->>KubeUnit: CreatePod() returns nil
+    else Error
+        KubeUnit->>KubeUnit: Handle error
+        alt ErrPodCompleted
+            KubeUnit->>KubeUnit: Check exit code
+            alt Exit code != 0
+                KubeUnit-->>KubeUnit: Return container failure error
+            else Exit code == 0
+                KubeUnit-->>KubeUnit: Return ErrPodCompleted
+            end
+        else Other error
+            KubeUnit->>KubeAPI: Try to get pod logs (fallback)
+            KubeUnit-->>KubeUnit: Return error with details
+        end
+    end
+```
+
+### Diagram 4: Logger Streaming Method (Recommended)
+
+This diagram shows the logger streaming method flow, including stdin streaming and log retrieval with reconnection support.
+
+```mermaid
+%%{init: {'theme':'base', 'themeVariables': { 'actorBkg':'#000000', 'actorBorder':'#4a9eff', 'actorTextColor':'#ffffff', 'actorLineColor':'#4a9eff', 'signalColor':'#ffffff', 'signalTextColor':'#ffffff', 'labelBoxBkgColor':'#000000', 'labelBoxBorderColor':'#4a9eff', 'labelTextColor':'#ffffff', 'loopTextColor':'#ffffff', 'noteBkgColor':'#000000', 'noteTextColor':'#ffffff', 'noteBorderColor':'#ff6b35', 'activationBkgColor':'#1a1a1a', 'activationBorderColor':'#4a9eff', 'sequenceNumberColor':'#000000'}}}%%
+sequenceDiagram
+    participant KubeUnit
+    participant KubeAPI as Kubernetes API
+    participant Pod
+    participant LogStream as Log Stream
+    participant StdoutFile as stdout file
+    participant StdinFile as stdin file
+
+    Note over KubeUnit, StdinFile: Logger streaming method workflow
+
+    KubeUnit->>KubeUnit: RunWorkUsingLogger()
+
+    alt Resume existing pod
+        KubeUnit->>KubeAPI: Get pod by name
+        KubeAPI-->>KubeUnit: Existing pod
+        Note over KubeUnit: skipStdin = true
+    else Create new pod
+        KubeUnit->>KubeAPI: CreatePod()
+        KubeAPI-->>KubeUnit: Pod created
+        Note over KubeUnit: skipStdin = false
```

Comment:
Can we add a note/sentence to explain why `skipStdin` would be `true`? I think it's when we connect to an existing pod, but it didn't jump out at me in the diagram.